### PR TITLE
Nc/playground

### PR DESCRIPTION
**Pull request by nfcampos**
- [x] Figure out why the frontend is retrying the request endlessly on error
- [x] Sort out the packaging of the js files in langserve releases
- [x] Sort out the env vars
- [ ] update development instructions in readme
- [x] Sort out the UI design
- [x] Make text input fields multi-line
- [ ] Sort out config defaults not applying in Safari

```
poetry run python examples/chain/server.py
cd langserve-playground && yarn dev
```

### CODE REVIEW

File: langserve-playground/src/useStreamLog.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+export function useStreamLog() {
+  const [latest, updateLatest] = useReducer(reducer, null);
+  const [controller, setController] = useState<AbortController | null>(null);
+
+  const startStream = useCallback(async (input: unknown, config: unknown) => {
+    const controller = new AbortController();
+    setController(controller);
+    await fetchEventSource("http://localhost:8003/stream_log", {
```
Comment (nfcampos) [inline langserve-playground/src/useStreamLog.tsx:54]: TODO

File: langserve-playground/src/useSchemas.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  const [schemas, setSchemas] = useState({
+    config: null,
+    input: null,
+  });
+
+  useEffect(() => {
+    async function save() {
+      if (import.meta.env.DEV) {
+        const [config, input] = await Promise.all([
+          fetch("http://localhost:8003/config_schema").then((r) => r.json()),
```
Comment (nfcampos) [inline langserve-playground/src/useSchemas.tsx:13]: TODO

File: langserve/playground.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    delimiter = "____"
+
+
+async def serve_playground(
+    runnable: Runnable, config_keys: list[str], base_url: str, file_path: str
+) -> Response:
+    local_file_path = os.path.join(
+        os.path.dirname(__file__),
+        "../langserve-playground/dist",
+        file_path or "index.html",
```
Comment (eyurtsev) [inline langserve/playground.py:20]: Is file path configurable to prototype with different variations of the UX?

**Review by eyurtsev — APPROVED** [review]

File: examples/chain/server.py [hunk 4]
```diff
@@ -1,34 +1,73 @@
...
+    "tell me a joke about {topic}.\nChat history: {chat_history}"
+).configurable_fields(
+    template=ConfigurableField(
+        id="prompt",
+        name="Prompt",
+        description="The prompt to use. Must contain {topic}",
+    )
+)
+chain = (
+    RunnablePassthrough.assign(chat_history=(lambda x: "\n".join(x)))
```
[the remark this answers is not in the document]
Comment (eyurtsev) [reply examples/chain/server.py:33]: ah it's fine didn't notice the custom input type

File: tests/unit_tests/test_server_client.py [hunk 5]
```diff
@@ -195,7 +196,7 @@ async def test_server_bound_async(app_for_config: FastAPI) -> None:
 
     # Test batch
     response = await async_client.post(
-        f"/h{config_hash}/batch",
+        f"/c/{config_hash}/batch",
```
[the remark this answers is not in the document]
Comment (nfcampos) [reply tests/unit_tests/test_server_client.py:199]: Yes I changed the url format for the configured endpoints

File: langserve/playground.py (hunk 3 above)
Comment (eyurtsev) [reply langserve/playground.py:30]: why is file path an input?

**Pull request merged**